### PR TITLE
Chronos job definition

### PR DESCRIPTION
**Pull request by maricaantonacci**

### CODE REVIEW

File: examples/indigo_job_output_upload_web.yaml [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+   output_protocol:
+     type: string
+     description: Protocol that will be used to upload the output generated by the job
+     constraints:
+        - valid_values: [ http, https, webdav, webdavs, ftp, ftps ]
+     required: yes
+
+   output_endpoint:
+     type: string
+     description: Endpoint to be used to upload the output generated by the job. Example: http://myhttpserver:8080/
```
Comment (micafer) [inline examples/indigo_job_output_upload_web.yaml]: @maricaantonacci you have to quote this description as it has the ":" char.

File: examples/indigo_job_output_upload_web.yaml [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+          ONEDATA_PROVIDERS: "ONEDATA_PROVIDERS_TO_BE_SET_BY_THE_ORCHESTRATOR"
+      artifacts:
+        image:
+          file: { get_input: docker_image }
+          type: tosca.artifacts.Deployment.Image.Container.Docker
+      requirements:
+        - host: docker_runtime1
+
+
+  chronos_job_upload:
```
Comment (micafer) [inline examples/indigo_job_output_upload_web.yaml]: @maricaantonacci, you  have to add two spaces before the chronos_job_upload node to have the correct alignement.

File: custom_types.yaml [hunk 3]
```diff
@@ -426,20 +426,28 @@ node_types:
...
         type: string
       retries:
         required: no
         type: integer
-      parents:
-        required: no
-        type: list
-        entry_schema:
-          type: string
+   capabilities:
```
Comment (micafer) [inline custom_types.yaml]: @maricaantonacci check the correct alignement. All this part is incorrectly aligned.

File: examples/indigo_job_onedata.yaml [hunk 4]
```diff
@@ -0,0 +1,126 @@
+tosca_definitions_version: tosca_simple_yaml_1_0
+
+imports:
+  - indigo_custom_types: https://raw.githubusercontent.com/indigo-dc/tosca-types/master/custom_types.yaml
+
+description: >
+  TOSCA examples for specifying a Chronos Job that runs an application using Onedata storage.
+
+inputs:
```
Comment (micafer) [inline examples/indigo_job_onedata.yaml]: @maricaantonacci the inputs section is under the topology_template section.

File: custom_types.yaml [hunk 5]
```diff
@@ -426,20 +426,28 @@ node_types:
...
+     - job_successor:
```
Comment (micafer) [inline custom_types.yaml]: @maricaantonacci you have to remove the "-" char before job_successor in the capabilities section.

File: examples/indigo_job_output_upload_web.yaml (hunk 1 above)
Comment (maricaantonacci) [inline examples/indigo_job_output_upload_web.yaml]: ok thanks

File: examples/indigo_job_output_upload_web.yaml (hunk 2 above)
Comment (maricaantonacci) [inline examples/indigo_job_output_upload_web.yaml]: fixed

File: custom_types.yaml (hunk 3 above)
Comment (maricaantonacci) [inline custom_types.yaml]: fixed

File: examples/indigo_job_onedata.yaml (hunk 4 above)
Comment (maricaantonacci) [inline examples/indigo_job_onedata.yaml]: fixed

File: custom_types.yaml (hunk 5 above)
Comment (maricaantonacci) [inline custom_types.yaml]: fixed

**Comment by maricaantonacci** [comment]
@micafer can you please help debugging the new check failures? Thank you!

**Comment by micafer** [comment]
@maricaantonacci now it is failing because the output_filenames input is not a string, as we discussed by email.

**Comment by maricaantonacci** [comment]
@micafer can you please paste the output of jenkins test?

File: custom_types.yaml [hunk 6]
```diff
@@ -426,20 +426,27 @@ node_types:
...
-        entry_schema:
-          type: string
+    capabilities:
+      job_successor:
+        type: tosca.capabilities.indigo.Container.Application.Docker.Chronos
+        valid_source_types: [tosca.nodes.indigo.Container.Application.Docker.Chronos]
+        occurrences: [0, UNBOUNDED]
+    requirements:
+      - job_predecessor:
+        capability: tosca.capabilities.indigo.Container.Application.Docker.Chronos
```
Comment (micafer) [inline custom_types.yaml]: @maricaantonacci lines 442 to 445 need one level more of indentation.

Comment (maricaantonacci) [inline custom_types.yaml]: Fixed

**Comment by micafer** [comment]
retest this please

File: examples/indigo_job_output_upload_swift.yaml [hunk 7]
```diff
@@ -0,0 +1,154 @@
...
+            num_cpus: { get_input: cpus }
+            mem_size: { get_input: mem }
+
+
+    docker_runtime2:
+      type: tosca.nodes.indigo.Container.Runtime.Docker
+      capabilities:
+        host:
+          properties:
+            num_cpus: 1
```
Comment (micafer) [inline examples/indigo_job_output_upload_swift.yaml]: @maricaantonacci it must be a float (1.0)

File: examples/indigo_job_output_upload_web.yaml [hunk 8]
```diff
@@ -0,0 +1,140 @@
...
+            num_cpus: { get_input: cpus }
+            mem_size: { get_input: mem }
+
+
+    docker_runtime2:
+      type: tosca.nodes.indigo.Container.Runtime.Docker
+      capabilities:
+        host:
+          properties:
+            num_cpus: 1
```
Comment (micafer) [inline examples/indigo_job_output_upload_web.yaml]: @maricaantonacci  it must be a float (1.0)

**Comment by maricaantonacci** [comment]
@micafer what is the problem now?

**Comment by micafer** [comment]
retest this please

**Comment by micafer** [comment]
Finally it was another issue in the tosca-parser ... I fixed it.

**Pull request merged**